tracing-log/src/lib.rs

```diff
@@ -91,6 +93,70 @@ pub trait AsTrace {
     fn as_trace(&self) -> Self::Trace;
 }
 
+pub trait GenericLogEvent<'a> {
+    fn normalized_metadata(&self, fields: &'a LogVisitor) -> Option<Metadata<'a>>;
+    fn is_log(&self) -> bool;
+}
+
+impl<'a> GenericLogEvent<'a> for Event<'a> {
+    fn normalized_metadata(&self, fields: &'a LogVisitor) -> Option<Metadata<'a>> {
+        if self.is_log() {
+            let original = self.metadata();
+
+            Some(Metadata::new(
+                "log event",
+                fields.target.as_ref().map(String::as_str).unwrap_or("log"),
+                original.level().clone(),
+                fields.file.as_ref().map(String::as_str),
+                fields.line.map(|l| l as u32),
+                fields.module_path.as_ref().map(String::as_str),
+                field::FieldSet::new(&[], self.metadata().callsite()),
+                Kind::EVENT,
+            ))
+        } else {
+            None
```

Comment:
As a thought, instead of returning an `Option`, we _could_ consider having this case just construct a new `Metadata` with the same values as the existing metadata. That might be less efficient, but would provide a nicer API...

Reply:
I don't like this `Option` either, but was indeed worried about efficiency. I'll try to change this. We may document the possible usage of `is_log` to allow filtering at the source while providing a clean interface?